packages/backend-function/src/factory.test.ts

```diff
@@ -236,4 +237,73 @@ void describe('AmplifyFunctionFactory', () => {
       },
     });
   });
+
+  void describe('runtime property', () => {
+    void it('sets valid runtime', () => {
+      const lambda = defineFunction({
+        entry: './test-assets/default-lambda/handler.ts',
+        runtime: 16,
+      }).getInstance(getInstanceProps);
+      const template = Template.fromStack(Stack.of(lambda));
+
+      template.hasResourceProperties('AWS::Lambda::Function', {
+        Runtime: Runtime.NODEJS_16_X.name,
+      });
+    });
+
+    void it('defaults to latest runtime', () => {
+      const lambda = defineFunction({
+        entry: './test-assets/default-lambda/handler.ts',
+      }).getInstance(getInstanceProps);
+      const template = Template.fromStack(Stack.of(lambda));
+
+      template.hasResourceProperties('AWS::Lambda::Function', {
+        Runtime: Runtime.NODEJS_LATEST.name,
+      });
+    });
+
+    void it('throws on deprecated runtime', () => {
+      assert.throws(
+        () =>
+          defineFunction({
+            entry: './test-assets/default-lambda/handler.ts',
+            runtime: 14,
+          }).getInstance(getInstanceProps),
+        new Error('runtime must be one of the following: 16, 18, 20')
+      );
+    });
+
+    void it('throws on invalid runtime', () => {
+      assert.throws(
+        () =>
+          defineFunction({
+            entry: './test-assets/default-lambda/handler.ts',
+            runtime: 100,
+          }).getInstance(getInstanceProps),
+        new Error('runtime must be one of the following: 16, 18, 20')
+      );
+    });
+
+    void it('throws on non-whole runtime number', () => {
+      assert.throws(
+        () =>
+          defineFunction({
+            entry: './test-assets/default-lambda/handler.ts',
+            runtime: 18.2,
+          }).getInstance(getInstanceProps),
+        new Error('runtime must be one of the following: 16, 18, 20')
+      );
+    });
+
+    void it('throws on negative runtime number', () => {
+      assert.throws(
+        () =>
+          defineFunction({
+            entry: './test-assets/default-lambda/handler.ts',
+            runtime: -18,
+          }).getInstance(getInstanceProps),
+        new Error('runtime must be one of the following: 16, 18, 20')
+      );
+    });
```

Comment:
these tests aren't testing a different codepath than the other test cases. Can be removed.